graylog2-web-interface/src/components/graylog/styles/menuItem.js

```diff
@@ -0,0 +1,44 @@
+import { css } from 'styled-components';
+
+import teinte from 'theme/teinte';
+import contrastingColor from 'util/contrastingColor';
+
+const menuItemStyles = (options = {}) => {
+  const { sibling = false } = options;
+
+  return css`
+  ${sibling && '& ~'} .dropdown-menu {
+    > li > a {
+      color: ${teinte.primary.tre};
+
+      :hover,
+      :focus {
+        color: ${contrastingColor(teinte.secondary.due)};
+        background-color: ${teinte.secondary.due};
+      }
+    }
+
+    > .active > a {
+      background-color: ${teinte.tertiary.due};
+      color: ${contrastingColor(teinte.tertiary.due)};
+
+      :hover,
+      :focus {
+        background-color: ${teinte.tertiary.uno};
+        color: ${contrastingColor(teinte.tertiary.uno)};
+      }
+    }
+
+    > .disabled > a {
+      background-color: ${teinte.secondary.due};
+      color: ${contrastingColor(teinte.secondary.due, 'AA')};
+
+      :hover {
+        color: ${contrastingColor(teinte.secondary.due, 'AA')};
+      }
+    }
+  }
+`;
+};
+
+export default menuItemStyles;
```

Comment:
What is your strategy for the styles defined in `components/graylog/styles/`? I always try to define the component related styles alongside the component. You are only extracting the color specific styles?

Reply:
It is being used on multiple components, so instead of repeating the code I am abstracting and importing it into each other place using the same styles